docs/ref/models/fields.txt

```diff
@@ -96,36 +96,42 @@ and the second element is the human-readable name. For example::
         ('SR', 'Senior'),
     )
 
-Generally, it's best to define choices inside a model class, and to
-define a suitably-named constant for each value::
+Django provides a special Enum type you can subclass to define this iterable
+in a concise way::
 
     from django.db import models
+    from django.utils.translation import ugettext as _
+
+    class YearInSchool(str, models.ChoiceEnum):
+        FRESHMAN = 'FR', _('Freshman')
+        SOPHOMORE = 'SO', _('Sophomore')
+        JUNIOR = 'JR', _('Junior')
+        SENIOR = 'SR', _('Senior')
+
 
     class Student(models.Model):
-        FRESHMAN = 'FR'
-        SOPHOMORE = 'SO'
-        JUNIOR = 'JR'
-        SENIOR = 'SR'
-        YEAR_IN_SCHOOL_CHOICES = (
-            (FRESHMAN, 'Freshman'),
-            (SOPHOMORE, 'Sophomore'),
-            (JUNIOR, 'Junior'),
-            (SENIOR, 'Senior'),
-        )
         year_in_school = models.CharField(
             max_length=2,
-            choices=YEAR_IN_SCHOOL_CHOICES,
-            default=FRESHMAN,
+            choices=YearInScool.choices,
+            default=YearInScool.FRESHMAN,
         )
 
         def is_upperclass(self):
-            return self.year_in_school in (self.JUNIOR, self.SENIOR)
+            return self.year_in_school in (YearInSchool.JUNIOR, YearInSchool.SENIOR)
+
+If you do not need to have the human-readable names translated, you
+can have them inferred from the enum member name (replacing
+underscores to spaces and using title-case)::
 
-Though you can define a choices list outside of a model class and then
-refer to it, defining the choices and names for each choice inside the
-model class keeps all of that information with the class that uses it,
-and makes the choices easy to reference (e.g, ``Student.SOPHOMORE``
-will work anywhere that the ``Student`` model has been imported).
+    class YearInSchool(str, models.ChoiceEnum):
+        FRESHMAN = 'FR'
+        SOPHOMORE = 'SO'
+        JUNIOR = 'JR'
+        SENIOR = 'SR'
+
+Since the case where the enum values need to be integers is extremely
+common, Django provides a ``ChoiceIntEnum`` class (inheriting int and
+``ChoiceEnum``).
```

Comment:
I would expect the case where enum values are strings to be similarly common. Perhaps we should provide `ChoiceStrEnum` too.

Reply:
We may. The real argument, actually, was mirroring the stdlib's `Enum` and `IntEnum`. More opinions welcome.